### ISSUE TITLE
Sparse IndexedSlices warning due to with tf.gather() and LossScaleOptimizer

### ISSUE THREAD

**Issue by thierryherrmann** [tensorflow/tensorflow#46459]
**System information**

    Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
    OS Platform and Distribution: Ubuntu 18.04.5 LTS
    TensorFlow installed from: binary
    TensorFlow version: 2.2.0
    Python version: 3.6.9
    CUDA/cuDNN version: 10.1.243 / 7.6.5
    GPU model and memory: NVidia Tesla V100-SXM2-32GB

**Describe the current behavior**
Use `tf.gather()` with `LossScalerOptimizer` triggers this warning:

`/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/indexed_slices.py:434: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.`

A fix (see colab notebook) using `tf.gather_nd()` removes the warning but breaks `saved_model_cli`.

**Describe the expected behavior**
The warning is unexpected (why does it appear only when using `LossScaleOptimizer` ?)  
The fix is not expected to trigger a `saved_model_cli` error

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/1vNX8ssp3nUTY_YOOml-ZfAa5yzPVHkl_?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
Possibly related issues:
- https://github.com/tensorflow/tensorflow/issues/38168
- https://github.com/tensorflow/tensorflow/issues/36948

**Comment by Saduf2019** [comment]
@thierryherrmann 
I ran the code shared on tf 2.2 and tf 2.4 and the mentioned issue does not occur on tf 2.4 please refer to the [gist here](https://colab.research.google.com/gist/Saduf2019/57f620adc81485c996c8b0eda7bf338b/untitled500.ipynb) and let us know.

**Comment by thierryherrmann** [comment]
Ok I've just looked at the gist which is a copy of the notebook I wrote but with contents duplicated for TF2.4.
Indeed the warning doesn't appear with TF2.4.
As mentioned in the **System Information**, I reported the issue against TF2.2 and I had not checked for any later version.
While I was at it I also just checked for TF2.3 and the issue doesn't occur for TF2.3 as well.
Can we hope for any fix for TF2.2 ?
Thanks

**Comment by chunduriv** [comment]
@thierryherrmann, Sorry for late response.

We see that you're using old version of TF2.2 and it is no longer supported. Its unlikely for TF2.2 version to receive any bug fixes except when we have security patches.  

As per [this comment](https://github.com/tensorflow/tensorflow/issues/46459#issuecomment-762335517), your issue was resolved in TF2.3 and TF2.4. Please feel free to move this to closed status. Thanks!

**Comment by thierryherrmann** [comment]
@chunduriv  yeah we no longer expected a response and switched to TF2.3 a long time ago. Thanks anyway.

**Comment by chunduriv** [comment]
@thierryherrmann, Please feel free to move this to closed status. Thanks!
